### DISCUSSION TITLE
Using a dedicated address for the sector-fault penalty

### DISCUSSION THREAD

**Discussion by TinySwitcher** [filecoin-project/FIPs#344] [Enhancements - Technical]
# Using a dedicated address for the sector-fault penalty

## Summary
Using a dedicated address for sector-fault penalties, the reward accounting procedure will be clear and convenient.

## Abstract
In the process of mining, it is reasonable that the service provider pays for the penalty when it happens, now it will be deducted from "Available balance" & "Locked rewards". Which will bring some significant trouble for account procedure.

## Change Motivation
According to [sector-faults](https://spec.filecoin.io/systems/filecoin_mining/sector/sector-faults/),  the penalty will happen when sector faults happened. And in the previous mechanism, it will be penalized from “Available balance" & "Locked rewards" (it depends on the time of sector faults happen). So, we want to create a dedicated address named "Penalty funds" to separate the penalty from rewards, then the reward accounting procedure will be clear and convenience

## Specification
1. clear the duty and responsibility, it is reasonable that the service provider pays for the penalty;
2. make the reward accounting procedure clear and convenient.

## Design Rationale
Add a separate account for the penalty in miner named "Penalty funds".

## Backwards Compatibility
1. when the penalty happens, lotus will deduct it from "Penalty funds" as a priority choice;
2. If the penalty funds are inefficient, then keep the previous sector fault penalty mechanism.

## Incentive Considerations
1. clear the duty and responsibility, it is reasonable that the service provider pays for the penalty;
2. make the reward accounting procedure clear and convenient;
3. separate the production accounting and business accounting.

## Product Considerations
separate the production accounting and business accounting.

## Implementation
Work in Progress.

**Comment by kaitlin-beegle** [comment]
Hello! FIP drafts need to be opened within the Discussion Forum, not as Issue.  I will convert this issue into a discussion; please let me know if you have any additional questions!

**Comment by anorth** [comment]
I'm not sure I understand the proposal. Let me have a go at restating what I think you're proposing:

- We have a separate account or actor which can be specified as a miner actor's first source for paying penalties. 
- If a miner is penalised, we first take funds from that actor to pay them. If it runs out, we revert to the actor's locked funds and balance.

This penalty address would be acting as a kind of guarantor for the miner, putting up some the money to protect the owner. But I infer that you don't want the miner worker/owner to otherwise have any control over those funds – the only way it can use them is to pay a penalty.

Am I on the right track? Could you be more specific about the situation in which this would be useful? What are the alternatives, and why are they undesirable?
